packages/commonwealth/server/util/cosmosProxy.ts

```diff
@@ -9,40 +10,65 @@ import { factory, formatFilename } from 'common-common/src/logging';
 const log = factory.getLogger(formatFilename(__filename));
 
 function setupCosmosProxy(app: Express, models: DB) {
-  // using bodyParser here because cosmjs generates text/plain type headers
-  app.post(
-    '/cosmosAPI/:chain',
-    bodyParser.text(),
-    async function cosmosProxy(req, res) {
-      log.trace(`Got request: ${JSON.stringify(req.body, null, 2)}`);
-      try {
-        log.trace(`Querying cosmos endpoint for chain: ${req.params.chain}`);
-        const chain = await models.Chain.findOne({
-          where: { id: req.params.chain },
-          include: models.ChainNode,
-        });
-        if (!chain) {
-          throw new AppError('Invalid chain');
-        }
-        log.trace(`Found cosmos endpoint: ${chain.ChainNode.url}`);
-        const response = await axios.post(chain.ChainNode.url, req.body, {
-          headers: {
-            origin: 'https://commonwealth.im',
-          },
-        });
-        log.trace(
-          `Got response from endpoint: ${JSON.stringify(
-            response.data,
-            null,
-            2
-          )}`
-        );
-        return res.send(response.data);
-      } catch (err) {
-        res.status(500).json({ message: err.message });
-      }
+  const getChain = async (req, res, next) => {
+    const chain = await models.Chain.findOne({
+      where: { id: req.params.chain },
+      include: models.ChainNode,
+    });
+    if (!chain) {
+      throw new AppError('Invalid chain');
     }
+    req.chainUrl = chain.ChainNode.url;
+    next();
+  };
+
+  app.use(
+    '/cosmosLCD/:chain',
+    getChain,
+    createProxyMiddleware({
+      target: 'https://api-eu-1.kyve.network',
+      // target: req.chainUrl, TODO: this doesn't work, why?
```

Comment:
@kurtisassad I can't figure out why this commented line won't work. It should be equivalent to the hard-coded line above. Something about setting it asynchronously is causing a problem.  

Any thoughts? Maybe I should try a different library?

Reply:
> @kurtisassad
> 
> > Do you think we can convert it to looking like this without breaking things?:
> 
> This would work too. We discussed it and common-common makes sense because that directory is meant to serve both commonwealth and chain-events.
> 
> In my most recent commit, I have pared the file count down and renamed the package `cosmos-ts`. It is now scoped only to `gov` v1 and its dependencies.

I think as a normal use case this would be fine. But you had turned it into its own package (with tsconfig, gitignore, npmrc, ...). 

If we decide to keep it in the common-common package, it shouldn't be its own sub-module. If we go the route of keeping it in common-common, these module config files should be removed. (Anyways, again we should wait on discussion from other devs before implementing this change).

Reply:
Do you have an entry for the chainUrl for this network in the ChainNodes table? Anyways for these more specific business logic questions, pinging me on slack might be better for faster response times.